### PR TITLE
Dark theme

### PR DESCRIPTION
**Pull request by gpx1000**
This will setup the dark theme based upon your browser preferences settings.

### CODE REVIEW

**Comment by SaschaWillems** [comment]
Gave this a quick spin, and it's already looking pretty nice. But before we get this out of the door we need to fix a few things. Right out of the box, there are elements that don't look good in dark mode.

Note: I only checked samples, guide and tutorial (not the spec) and also only did a quick runthrough.

E.g. the top bar:

image

The search bar doesn't match the dark theme, and the navigation links are black on dark red, which is pretty low in contrast, making them very hard to read. 

Also the search term is almost impossible to read with dark gray on darker gray, esp. as the result popover is still white:

image

Same for the breadcrumb, which is gray on gray and very hard to read:

image

Bottom navigation also needs fixing:

image

Same for pages with right hand navigation:

image

Source highlighting also needs to be adjusted, as it currently has some very low contrast elements:

image

We also need to find a way to deal with transparent images that are pretty much unreadable now, like this from the Index buffer chapter of the tutorial:

image

Or the start page of the guide:

image

We should also adjust the link color, as the dark blue on black doesn't have a good contrast or visibility:

image



To sum it up: Great first step, but those tiny bits to make it work and look good for all content is probably gonna take quite some time and effort ;)

**Comment by SaschaWillems** [comment]
Maybe we can take some queues from https://docs.spring.io/spring-framework/reference/overview.html

They also use Antora and have a dark mode toggle.

The site also seems to do some additional things that could be interesting for our docs site.

**Comment by SaschaWillems** [comment]
Maybe we could take some queues from dark mode extensions (e.g. Dark reader for Chrome). This is what the site looks for me with that extension:

docs vulkan org_tutorial_latest_03_Drawing_a_triangle_00_Setup_00_Base_code html

Not perfect but very decent. It's using a color scheme similar to what I use on https://www.gpuinfo.org/

Dark mode seems to be the no 1 request for our docs site btw. So maybe we could prioritize this a bit ;)

**Comment by oddhack** [comment]
Note that #16 will create a (very minor) conflict with this PR when it is merged.

**Comment by oddhack** [comment]
On today's docs telecon, I suggested using the ANDI tool to evaluate the dark mode theme for accessibility issues (this is not to say there are no issues with the regular theme, either). ANDI has a bunch of different categories of which the "graphics/images" and "color contrast" are probably most relevant to dark mode. The other categories (appear to be) independent of dark / light mode theme so would probably not benefit from testing specifically for dark mode.

**Comment by SaschaWillems** [comment]
Added backgrounds to all svg images in the tutorial, see https://github.com/KhronosGroup/Vulkan-Tutorial/commit/5f36bc08921bea50f6dfab07a5b26d27bc5a2ec0

Without this change, those images would no longer be properly visible with dark mode.

Not the perfect solution, might try to make the svgs adjust to the selected theme at some point.

**Comment by marty-johnson59** [comment]
Awesome, thanks!
